tasks/tasks.go

```diff
@@ -38,12 +40,37 @@ func LoadStorageEngine(ctx context.Context, config *tsbridge.Config) (storage.Ma
 	}
 }
 
-// Sync updates all configured metrics.
-func Sync(ctx context.Context, config *tsbridge.Config, metrics *tsbridge.Metrics, store storage.Manager) error {
-	metricCfg, err := tsbridge.NewMetricConfig(ctx, config, store)
+// Checks if the metric config file has been updated, and updates the cached file stats if it has.
+func metricConfigFileUpdated(ctx context.Context, config *tsbridge.Config, metricCfgFs *os.FileInfo) (bool, error) {
+	currMetricCfgFs, err := os.Stat(config.Options.Filename)
+	if err != nil {
+		return false, err
+	}
+	if (*metricCfgFs).ModTime() != currMetricCfgFs.ModTime() && (*metricCfgFs).Size() != currMetricCfgFs.Size() {
+		return true, nil
+	}
+	return false, nil
+}
+
+// SyncMetricConfig ensures that metric config is always synced with the metric config file. This should only be called when !env.IsAppEngine().
+func SyncMetricConfig(ctx context.Context, config *tsbridge.Config, store storage.Manager, metricCfg *tsbridge.MetricConfig) error {
```

Comment:
In all these funcs, you're passing the same values over. the config struct, metric config etc.

Would it be cleaner to just store this in a struct? (please evaluate and see what's the best option here)

```
type Sync struct {
 cfg *tsbridge.Config
 metricCfg *tsbridge.MetricConfig
}
```

Then you can just call something like:

```
func (s *Sync) Config(ctx context.Context, store store.Manager) error {
 update, err := isCfgChanged(ctx)
....
}
```

This way you avoid passing the vars / pointers around and makes it easier to read.

Reply:
As discussed offline, this would require a lot of refactoring across `tasks.go` and `handlers.go`. To keep this PR small, I created an issue leave this refactor for future work: https://github.com/google/ts-bridge/issues/104

Reply:
Sounds good.